v2/connection/connection_http_internal.go

```diff
@@ -276,3 +266,18 @@ func (j httpConnection) do(ctx context.Context, req *httpRequest) (*httpResponse
 
 	return &httpResponse{response: resp, request: req}, nil, nil
 }
+
+// getDecoderByContentType returns the decoder according to the content type.
+// If content type is unknown then nil is returned.
+func getDecoderByContentType(contentType string) Decoder {
+	switch contentType {
+	case ApplicationVPack:
+		return getVPackDecoder()
+	case ApplicationJSON:
+		return getJsonDecoder()
+	case PlainText, ApplicationOctetStream, ApplicationZip:
```

Comment:
maybe it should be the default decoder? if it is not the VPack and JSON then fetch raw bytes?